glean-core/android/src/test/java/mozilla/telemetry/glean/DispatchersTest.kt

```diff
@@ -178,4 +178,50 @@ class DispatchersTest {
             assertEquals("This list is out of order $orderedList", num, orderedList[num])
         }
     }
+
+    @Test
+    fun `dispatched tasks throwing exceptions are correctly handled`() {
+        val mainThread = Thread.currentThread()
+        val threadCanary = AtomicInteger()
+
+        // By setting testing mode to false, we make sure that things
+        // are executed asynchronously.
+        Dispatchers.API.setTestingMode(false)
+        Dispatchers.API.setTaskQueueing(false)
```

Comment:
One thing I stumbled upon before while reading those tests: do we need to reset those values after the tests?

Reply:
Mh, for this specific file, I don't think it's necessary. Each test encodes verifies assumptions before executing the rest. For other tests, it might be a good thing to verify/set in `resetGlean`, but this might have implications. I filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1615570) for auditing.